pkg/security/resolvers/dentry/resolver.go

```diff
@@ -532,17 +532,30 @@ func (dr *Resolver) ResolveFromERPC(pathKey model.PathKey, cache bool) (string,
 
 	var keys []model.PathKey
 	var entries []PathEntry
+	var nextInode uint64
+	var nextMountID uint32
+	i := 0
 
 	filenameParts := make([]string, 0, 128)
 
-	i := 0
+	nextInode = model.ByteOrder.Uint64(dr.erpcSegment[i : i+8])
+	if nextInode != pathKey.Inode {
+		dr.missCounters[entry].Inc()
+		return "", fmt.Errorf("unable to resolve the path of mountID `%d` and inode `%d` with eRPC: %w", pathKey.MountID, pathKey.Inode, err)
+	}
+	nextMountID = model.ByteOrder.Uint32(dr.erpcSegment[i+8 : i+12])
+	if nextMountID != pathKey.MountID {
+		dr.missCounters[entry].Inc()
+		return "", fmt.Errorf("unable to resolve the path of mountID `%d` and inode `%d` with eRPC: %w", pathKey.MountID, pathKey.Inode, err)
```

Comment:
I would suggest to create a custom error. We have 3 times the same message and it would help to catch the kind of error later in the code.